manifests/app.pp

```diff
@@ -103,8 +103,9 @@
       active  => true,
     }
     ~> service { "unicorn_${name}":
-      ensure => running,
-      enable => true,
+      ensure  => running,
+      enable  => true,
+      restart => "/bin/systemctl reload-or-restart unicorn_${name}.service",
```

Comment:
I think when we migrated to using the modern systemd module, we overlooked that it also handles notifying the service [here](https://github.com/voxpupuli/puppet-systemd/blob/3.0.0/manifests/unit_file.pp#L105), so we should be able to change the [chaining arrow](https://puppet.com/docs/puppet/7/lang_relationships.html#lang_rel_chaining_arrows) from `~>` to `->`, which I believe is causing the double restart.

```puppet
systemd::unit_file { "unicorn_${name}.service":
  content => template('unicorn/systemd/unicorn.service.erb'),
  owner   => 'root',
  group   => '0',
  mode    => '0644',
  enable  => true,
  active  => true,
  restart => "/bin/systemctl reload-or-restart unicorn_${name}.service",
}
-> service { "unicorn_${name}":
  ensure  => running,
  enable  => true,
}
```